### PR TITLE
feat: added `cspell`, custom paths, lint staged files & updated all npm packages

### PR DESCRIPTION
**Pull request by IgnatG**
### Changelog
- added `lint-staged` (this will lint staged files)
- order packages inside `package.json` automatically, via `sort-package-json`
- updated all npm packages
- added cspell (to check for spelling mistakes)
- updated the linting rules and converted the `.js` file to `.json` (to keep the file names consistent)
- added custom paths via `tsconfig.json`

### CODE REVIEW

File: src/modules/markets/entities/market.entity.ts [hunk 1]
```diff
@@ -59,11 +59,11 @@ export class MarketEntity {
 
   @Column({ name: 'ESRA_ID' })
   @ApiProperty({ example: 1, description: '' })
-  ESRAClasificationId: number;
+  ESRAClassificationId: number;
```
Comment (IgnatG) [inline src/modules/markets/entities/market.entity.ts:62]: @avaitonis  can you check if this is correct? there was a typo here

File: src/modules/markets/entities/market.entity.ts [hunk 2]
```diff
@@ -59,11 +59,11 @@ export class MarketEntity {
...
 
   @Column({ name: 'ESRA_DESC' })
   @ApiProperty({ example: 'Standard Risk', description: '' })
-  ESRAClasificationDesc: string;
+  ESRAClassificationDesc: string;
```
Comment (IgnatG) [inline src/modules/markets/entities/market.entity.ts:66]: same as above, @avaitonis  can you check if this is correct? there a typo here

**Review by abhi-markan — APPROVED** [review]

File: src/modules/markets/entities/market.entity.ts (hunk 1 above)
Comment (avaitonis) [reply src/modules/markets/entities/market.entity.ts:62]: This typo was introduced 7 months ago and endpoint is now used by EXIP. We can fix typo but release of this nodeJS API will require small change in EXIP project.

File: src/modules/markets/entities/market.entity.ts (hunk 2 above)
Comment (avaitonis) [reply src/modules/markets/entities/market.entity.ts:66]: This typo was introduced 7 months ago and endpoint is now used by EXIP. We can fix typo but release of this nodeJS API will require small change in EXIP project.

**Review by avaitonis — APPROVED** [review]

File: src/modules/markets/entities/market.entity.ts (hunk 1 above)
Comment (IgnatG) [reply src/modules/markets/entities/market.entity.ts:62]: ok cool. let's fix it as part of this PR and we will notify Tony once the API is live

**Pull request merged**